packages/react-reconciler/src/ReactFiber.old.js

```diff
@@ -851,6 +853,7 @@ export function assignFiberPropertiesInDEV(
   target.pendingProps = source.pendingProps;
   target.memoizedProps = source.memoizedProps;
   target.updateQueue = source.updateQueue;
+  target.memoCache = source.memoCache;
```

Comment:
Since this field is only relevant to functional components, I suggest putting this field on a different object that only exists for hooks.

I think the best place is the FunctionalComponentUpdateQueue object. The existing fields on that object are conceptually similar because they, too, are copy-on-write. For example, this is how we track usages of `useSyncExternalStore`:

https://github.com/facebook/react/blob/aa80a309b19d59832ffeb82dfb6235da0c93dd42/packages/react-reconciler/src/ReactFiberHooks.new.js#L1502-L1513

Reply:
It's possible we may eventually want to move this to a fiber-level field if we end up using the Forget cache for non-user components, too, but since that's only theoretical for now I think it's best to move it to reduce the memory impact.

Reply:
You had suggested that but the updateQueue property is [set to null prior to calling the render function](https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberHooks.new.js#L399), so it isn't clear to me how we'd make the cache value available to `useMemoCache()`. Do you mean to add a new global that we set/unset as we enter/exit each component?

Reply:
You can read the previous values from the current fiber: `current.updateQueue`